qi-lib/flow/compiler.rkt

```diff
@@ -473,32 +483,42 @@ the DSL.
     (syntax-parse stx
       [(_ consequent:clause
           alternative:clause)
-       #'(λ (f . args)
-           (if (apply f args)
-               (apply (qi0->racket consequent) args)
-               (apply (qi0->racket alternative) args)))]
+       #'(let ([compiled-if-flow
+                (λ (f . args)
+                  (if (apply f args)
+                      (apply (qi0->racket consequent) args)
+                      (apply (qi0->racket alternative) args)))])
+           compiled-if-flow)]
       [(_ condition:clause
           consequent:clause
           alternative:clause)
-       #'(λ args
-           (if (apply (qi0->racket condition) args)
-               (apply (qi0->racket consequent) args)
-               (apply (qi0->racket alternative) args)))]))
+       #'(let ([compiled-if-flow
+                (λ args
+                  (if (apply (qi0->racket condition) args)
+                      (apply (qi0->racket consequent) args)
+                      (apply (qi0->racket alternative) args)))])
+           compiled-if-flow)]))
 
   (define (fanout-parser stx)
     (syntax-parse stx
       [_:id #'repeat-values]
+      [(_ 0) #'*->1]
+      [(_ 1) #'values]
```

Comment:
I'm not able to verify the original behavior here at the moment, but does this PR modify the behavior of any edge cases, e.g. `(fanout 0)` or `(gen)`, or `(relay)`? If it does, it would be great to add tests for these cases. For cases like `(fanout 0)` and `(fanout 1)` we would need tests even if the behavior hasn't changed since it would now hit different code that needs to be covered by tests (unfortunately the coverage check on PRs [doesn't work](https://github.com/countvajhula/qi/issues/48) at the moment, but you can run `make cover` to generate a coverage report locally).

Reply:
This PR doesn't modify the original behavior, I will write more tests later.